magefile.go

```diff
@@ -65,26 +65,53 @@ func Build() error {
 }
 
 func fetchPackageStorage() error {
-	err := os.RemoveAll(storageRepoDir)
+
+	// If storage directory does not exists, check it out
+	if _, err := os.Stat(storageRepoDir); os.IsNotExist(err) {
+		err = sh.Run("git", "clone", "--depth=1", "https://github.com/elastic/package-storage.git", storageRepoDir)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		// catch other errors
+		return err
+	}
+
+	packageStorageRevision := os.Getenv("PACKAGE_STORAGE_REVISION")
+	if packageStorageRevision == "" {
+		packageStorageRevision = "master"
+	}
+
+	err := sh.Run("git",
+		"--git-dir", filepath.Join(storageRepoDir, ".git"), "--work-tree", storageRepoDir,
+		"reset", "--hard")
 	if err != nil {
 		return err
 	}
 
-	err = sh.Run("git", "clone", "https://github.com/elastic/package-storage.git", storageRepoDir)
+	err = sh.Run("git",
+		"--git-dir", filepath.Join(storageRepoDir, ".git"), "--work-tree", storageRepoDir,
+		"clean", "-df")
```

Comment:
I'm not sure about this change. I used to fetch the repository to build directory and do not touch it. If I need to play with it, I would go to the package-storage. Do we need this adjustment?

Reply:
It's not a requirement. It just makes sure in case someone modified the build directory it will be reset / cleaned up.